hangman/game.py

```diff
@@ -1,13 +1,116 @@
 from .exceptions import *
-
+import random
 
 class GuessAttempt(object):
-    pass
+    
+    def __init__(self, guess, hit=None, miss=None):
+        self.guess = guess
+        self.hit = hit
+        self.miss = miss
+    
+        if self.hit == True and self.miss == True:
+            raise InvalidGuessAttempt
+    
+    def is_hit(self):
+        if self.hit == None:
+            return False
+        else:
+            return True
+
+    def is_miss(self):
+        if self.miss == None:
+            return False
+        else:
+            return True
 
 
 class GuessWord(object):
-    pass
+    
+    def __init__(self, answer):
+        self.answer = answer
+        self.masked = '*'*len(self.answer)
 
+        if len(self.answer) == 0:
+            raise InvalidWordException
+    
+    def perform_attempt(self, guess):
+        if len(guess) > 1:
+            raise InvalidGuessedLetterException
+
+        guess = guess.lower()
+        self.answer = self.answer.lower()
+        
+        if guess in self.answer:
+            result = ''
+            index = 0
+            for i in self.answer:
+                if i.lower() == guess:
+                    result += guess
+                else:
+                    result += self.masked[index]
+                index += 1
+            self.masked = result
+        
+        if guess in self.answer:
+            attempt = GuessAttempt(guess, hit=True)
```

Comment:
this can be done in the same block above since it's the same condition ... `if guess in self.answer` exists in line 43